packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -268,15 +273,38 @@ export class SummaryGenerator {
 		);
 
 		let summaryData: SubmitSummaryResult | undefined;
+
+		// Called when summarization fails at any stage. Validates that the right params are passed when a particular
+		// stage failed. Logs Summarize_cancel event.
 		const fail = (
 			errorCode: keyof typeof summarizeErrors,
 			error?: any,
+			submitFailureResult?: ISubmitSummaryFailureResult,
+			nackFailureResult?: INackSummaryResult,
 			properties?: SummaryGeneratorTelemetry,
-			nackSummaryResult?: INackSummaryResult,
 		) => {
-			// UploadSummary may fail with 429 and retryAfter - respect that
-			// Summary Nack also can have retryAfter, it's parsed below and comes as a property.
-			const retryAfterSeconds = getRetryDelaySecondsFromError(error);
+			let retryAfterSeconds: number | undefined;
+			switch (errorCode) {
+				case "submitSummaryFailure":
+					assert(
+						submitFailureResult !== undefined,
```

Comment:
The comment on the summarizeErrors case for "submitSummaryFailure" says:

> Error encountered while generating the summary tree, uploading it to storage, or submitting the op.

Sounds to me like that encompasses both `submitFailureResult` and `broadcastFailureResult` right?

Reply:
Never mind, now I see that broadcast is about when we see Summary op ack'd (broadcast)